mailman/submit.py

```diff
@@ -39,9 +39,32 @@ def main(argv):
 
     # if the 'cover letter' is not addressed to
     # an assignment inbox, this email session
-    # isn't a patchset at all
+    # isn't a patchset at all, but it could be a peer review.
     if cover_letter.rcpt not in ASSIGNMENT_LIST:
-        # TODO process peer review
+        if patches:
+            return 0
+        with open(f'/var/lib/email/mail/{cover_letter.msg_id}', 'r') as f:
+            for line in f:
+                if line == '':
+                    return 0
+                if line.startswith('In-Reply-To: '):
+                    lt_ind = line.find('<')
+                    at_ind = line.find('@')
+                    if lt_ind == -1 or at_ind == -1 or lt_ind > at_ind:
+                        return 0
+                    irt = line[lt_ind+1:at_ind]
+                    break
+        # 'clear the lower 16 bits' to get the reviewee patchset id
+        patchset_id = irt[:-4]+'0000'
+        assignment = (db.Submission
+                        .select()
+                        .where(db.Submission.submission_id == patchset_id)
+                        .first()).assignment
+        if assignment not in ASSIGNMENT_LIST:
+            return 0
```

Comment:
This should be unnecessary since it only goes into the DB if it was a valid assignment right?

Reply:
If we're going to let this spit error messages we might as well skip this check too right?
```
if lt_ind == -1 or at_ind == -1 or lt_ind > at_ind:
                        return 0
```
(mailman/submit.py lines 53 and 54)
(I mean I can't imagine this check would ever trip regardless, I put it in as a just-in-case.)

Reply:
fair point -  if either of them is not found worst case you just get some garbled or empty substring because `-1` is valid to use in slicing, it can't even crash, though this feels like more of a sane check to me than verifying that the data we already have in the DB is valid but you can remove both. Whatever you think is right